### PR TITLE
fix: various unitinialized or badly initialized variables

### PR DESCRIPTION
**Pull request by nerzhul**
Also check some return values as proposed by coverity, if they are relevant to be triggered as a bug

## To do

This PR is a Ready for Review.

### CODE REVIEW

**Comment by nerzhul** [comment]
@sfan5 @rubenwardy can you take a look at this pretty simple bugfix one ? 
Thanks

File: src/filesys.cpp [hunk 1]
```diff
@@ -856,6 +856,12 @@ bool extractZipFile(io::IFileSystem *fs, const char *filename, const std::string
 }
 #endif
 
+/**
+ * Read arbitrary file and return read status
+ * @param path Path on filesystem
+ * @param out string reference to write to
+ * @returns read success status
+ */
```
Comment (SmallJoker) [inline src/filesys.cpp:864]: I'd rather put Doxygen comments into headers, as a convention.

File: src/client/client.cpp [hunk 2]
```diff
@@ -103,6 +103,7 @@ Client::Client(
 		GameUI *game_ui,
 		ELoginRegister allow_login_or_register
 ):
+	m_allow_login_or_register(allow_login_or_register),
```
Comment (SmallJoker) [inline src/client/client.cpp:106]: This variable should be private. There are no accesses outside of the Client class.

File: src/mapgen/mg_biome.h [hunk 3]
```diff
@@ -247,6 +247,6 @@ class BiomeManager : public ObjDefManager {
 private:
 	BiomeManager() {};
 
-	Server *m_server;
+	Server *m_server = nullptr;
```
Comment (SmallJoker) [inline src/mapgen/mg_biome.h:250]: FYI: Superfluous due to initializer list in the source file. This is a comment and not necessary a change request.

File: src/gui/guiFormSpecMenu.h [hunk 4]
```diff
@@ -259,7 +259,7 @@ class GUIFormSpecMenu : public GUIModalMenu
 	void acceptInput(FormspecQuitMode quitmode=quit_mode_no);
 	bool preprocessEvent(const SEvent& event);
 	bool OnEvent(const SEvent& event);
-	bool doPause;
+	bool doPause = false;
```
Comment (SmallJoker) [inline src/gui/guiFormSpecMenu.h:262]: Makes the additional line in `guiFormSpecMenu.cpp`/`GUIFormSpecMenu::create` superfluous

**Pull request closed without merging**